src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -40,416 +44,554 @@
+    }
+
+    @Override
+    public String[] getDependency() {
+        return null;
+    }
+
+    @Override
+    public String getDescription() {
+        if (vuln != null) {
+            return vuln.getDescription();
+        }
+        return "Failed to load vulnerability description from file";
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.SERVER;
+    }
+
+    @Override
+    public String getSolution() {
+        if (vuln != null) {
+            return vuln.getSolution();
+        }
+        return "Failed to load vulnerability solution from file";
+    }
+
+    @Override
+    public String getReference() {
+        if (vuln != null) {
+            StringBuilder sb = new StringBuilder();
+            for (String ref : vuln.getReferences()) {
+                if (sb.length() > 0) {
+                    sb.append('\n');
+                }
+                sb.append(ref);
+            }
+            return sb.toString();
+        }
+        return "Failed to load vulnerability reference from file";
+    }
+    @Override
+    public void init() {
+        if (INSECURE_METHODS == null) {
+            INSECURE_METHODS = new ArrayList<String>();
+            INSECURE_METHODS.addAll(Arrays.asList(INSECURE_DEFAULT_METHODS));
+            INSECURE_METHODS.addAll(Arrays.asList(WEBDAV_METHODS));
+        }
+    }
+
+    @Override
+    public void scan() {
+
+        try {
+            String allowedmethods = null;
+            String publicmethods = null;
+            String thirdpartyHost = "www.google.com";
+            int thirdpartyPort = 80;
+            Pattern thirdPartyContentPattern = Pattern.compile("<title.*Google.*/title>", Pattern.CASE_INSENSITIVE);
+
+            AttackStrength attackStrength = getAttackStrength();
+            if (attackStrength == AttackStrength.HIGH || attackStrength == AttackStrength.INSANE) {
+                // in this case, we do not bother with the OPTIONS method, but
+                // try all the insecure methods on the URL directly
+                // this is useful in the case where the OPTIONS method does not
+                // report the method, but where it is actually supported
+                // in this case, if a vulnerability is reported, there is little
+                // doubt that it is real
+
+                // try the TRACK method
+                testTraceOrTrack(this.getBaseMsg(), HttpRequestHeader.TRACK);
+                // try the TRACE method
+                testTraceOrTrack(this.getBaseMsg(), HttpRequestHeader.TRACE);
+
+
+                // use a CONNECT method to try establish a socket connection to
+                // a third party, via the server being tested
+                testConnect(this.getBaseMsg(), thirdpartyHost, thirdpartyPort, thirdPartyContentPattern);
+
+                /* Test all other methods */
+                for (String insecureMethod : INSECURE_METHODS) {
```

Comment:
The list contains `TRACK`, `TRACE`, and `CONNECT` so they are being checked again?

Reply:
They were being ignored in testHttpMethod method but it was unnecessary, I fixed it.